src/main/java/bio/terra/janitor/db/JanitorDao.java

```diff
@@ -69,16 +69,110 @@ public TrackedResourceId createResource(
               .map(
                   entry ->
                       new MapSqlParameterSource()
-                          .addValue("tracked_resource_id", trackedResourceId.id())
+                          .addValue("tracked_resource_id", resource.trackedResourceId().uuid())
                           .addValue("key", entry.getKey())
                           .addValue("value", entry.getValue()))
-              .toArray(size -> new MapSqlParameterSource[labels.size()]);
+              .toArray(MapSqlParameterSource[]::new);
 
       jdbcTemplate.batchUpdate(insertLabelSql, sqlParameterSourceList);
     }
-    return trackedResourceId;
   }
 
+  /**
+   * Retrieves and updates a TrackedResource that is ready and has expired by {@code now} to {@link
+   * TrackedResourceState#CLEANING}. Inserts a new {@link CleanupFlight} for that resource as
+   * initiating.
+   */
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public Optional<TrackedResource> updateResourceForCleaning(Instant now, String flightId) {
```

Comment:
I didn't see how we use this. But here, do we want to do 1 by 1 or by `limit` at one time?  Seems currently it only process 1 record for each call, is the efficient?  And if the external method keep calling it, will they increase `now` for each call?

Reply:
This is doing it 1 by 1. Each Stairway submission is going to be 1 by 1, so it's a little moot to worry about it. If we have performance issues, we can adjust.
It's up to the caller to adjust `now`. Getting the system time is going to be much smaller than network connection to the database. Or the caller can use a single `now` in repeated calls to just get things that had expired when they started and get the rest the next go around.

Yeah, these are not being used yet. Downside of breaking up PRs into smaller chunks is the whole picture isn't there at once. I can upload some WIP use for a look if you like.

Reply:
(1) 1 by 1 make sense then if that is what stariway needs
(2) If the `now` is the adjustable, wdyt not to use now term, and this method would be updateRecordForCleaning at certain timestamp, and the timestamp can be any things depends what upstream system needs?
(3) WIP pr is not needed. We can always adjust the code here in follow PRs where it is used.

Reply:
2) Open to different parameter names. I think `now` is reasonable because it captures the intent - what is ready to clean depends on what time it is "now" as determined by the caller. You should be calling this with a notion of "now," not next week. Brainstorming other parameters names: `expirationAtLeast`, `expiredBy`, other ideas?

Reply:
expiredBy or the other SG to me. 
I think the logic for this Dao's method doesn't tight to ''now;' context. Just from reading the code and understanding the logic, seems it just updateRecordForCleaning by the time and flight id whatever upstream layer gives to it.

Reply:
`expiredBy` done.